machineconfiguration/v1alpha1/types_pinnedimageset.go

```diff
@@ -0,0 +1,83 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PinnedImageSet describes a set of images that should be pinned by CRI-O and pulled to the nodes matching the machine config pool selector.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type PinnedImageSet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec describes the configuration of this pinned image set.
+	// +kubebuilder:validation:Required
+	Spec PinnedImageSetSpec `json:"spec"`
+	// status describes the last observed state of this pinned image set.
+	// +optional
+	Status PinnedImageSetStatus `json:"status"`
+}
+
+// PinnedImageSetSpec defines the desired state of a PinnedImageSet.
+type PinnedImageSetSpec struct {
+	// MachineConfigPoolSelector selects which pools the PinnedImageSet should apply to.
+	// A nil selector will result in no pools being selected.
+	// +optional
+	MachineConfigPoolSelector *metav1.LabelSelector `json:"machineConfigPoolSelector,omitempty"`
```

Comment:
Moving to MCP selecting a set of pinned image lists?

Reply:
correct as per conv offline/above.